lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
```

Comment:
I'm curious if you have tried comparing the content from the model without converting it. `draft-js` is an immutable-based system and so I would think that we could accomplish this by a simple reference check…

```js
const hasChangedContent = editorState.getCurrentContent() !== this.state.editorState.getCurrentContent();
```

also I have rearranged the variable name to keep the boolean-ish verb out front